core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -1272,6 +1272,10 @@ private boolean sameTypeOrNarrowsNullability(RelDataType oldType, RelDataType ne
       return false;
     }
 
+    @Override public Boolean visitNamedParam(RexNamedParam namedParam) {
```

Comment:
I think it might be valid to have this be true? IE, I think we want CASE WHEN @a == 1 THEN TRUE WHEN @a == 2 THEN TRUE ELSE FALSE to become. (@a == 1) | (@a == 2)

It's probably safer for this to be false for right now though.

Reply:
I'll file a JIRA to investigate this later.

Reply:
Here is the JIRA: https://bodo.atlassian.net/browse/BS-533